File: src/main/java/com/ptaf/ai/validation/PageFrameContextGuardResult.java
package com.ptaf.ai.validation;

import java.util.List;

public record PageFrameContextGuardResult(
        boolean passed,
        List<String> invalidFrameSteps,
        List<String> invalidPageSteps,
        List<String> warnings,
        List<String> blockingErrors,
        int frameStepCount,
        int pageStepCount
) {
}

File: src/main/ptaf/ai/validation/page_frame_context_guard_result.py
from __future__ import annotations

from dataclasses import dataclass


@dataclass(frozen=True)
class PageFrameContextGuardResult:
    passed: bool
    invalid_frame_steps: list[str]
    invalid_page_steps: list[str]
    warnings: list[str]
    blocking_errors: list[str]
    frame_step_count: int
    page_step_count: int
